File: Widget/Objective-C/Widget Objective-C/FirstVC.m
//
//  FirstVC.m
//  Widget Objective-C
//
//  Created by pro648 on 18/7/22
//  Copyright © 2018年 pro648. All rights reserved.
//

#import "FirstVC.h"
#import <NotificationCenter/NotificationCenter.h>

@interface FirstVC ()

@end

@implementation FirstVC

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)handleButtonTapped:(UIButton *)sender {
    if ([sender.titleLabel.text isEqualToString:@"Show Widget"]) {
        [[NCWidgetController widgetController] setHasContent:YES forWidgetWithBundleIdentifier:@"i.Widget-Objective-C.UsedSpaceWidget"];
        [sender setTitle:@"Hide Widget" forState:UIControlStateNormal];
    } else {
        [[NCWidgetController widgetController] setHasContent:NO forWidgetWithBundleIdentifier:@"i.Widget-Objective-C.UsedSpaceWidget"];
        [sender setTitle:@"Show Widget" forState:UIControlStateNormal];
    }
}

@end

File: Widget/Swift/Widget Swift/FirstVC.swift
//
//  FirstVC.swift
//  Widget Swift
//
//  Created by pro648 on 18/7/22
//  Copyright © 2018年 pro648. All rights reserved.
//

import UIKit
import NotificationCenter

class FirstVC: UIViewController {
    
    @IBOutlet weak var label: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func handleButtonTapped(_ sender: UIButton) {
        if sender.titleLabel?.text == "Show Widget" {
            NCWidgetController().setHasContent(true, forWidgetWithBundleIdentifier: "pro648.Widget-Swift.UsedSpaceWidget")
            sender.setTitle("Hide Widget", for: .normal)
        } else {
            NCWidgetController().setHasContent(false, forWidgetWithBundleIdentifier: "pro648.Widget-Swift.UsedSpaceWidget")
            sender.setTitle("Show Widget", for: .normal)
        }
    }
}
